types/leaflet/index.d.ts

```diff
@@ -728,10 +734,13 @@ export class LayerGroup<P = any> extends Layer {
     feature?: geojson.FeatureCollection<geojson.GeometryObject, P> | geojson.Feature<geojson.MultiPoint, P> | geojson.GeometryCollection;
 }
 
+// @factory L.layerGroup(layers?: Layer[], options?: Object)
 /**
- * Create a layer group, optionally given an initial set of layers.
+ * Create a layer group, optionally given an initial set of layers and an `options` object.
  */
-export function layerGroup(layers: Layer[]): LayerGroup;
+export function layerGroup(layers?: Layer[]): LayerGroup;
+// tslint:disable-next-line
+export function layerGroup(layers: Layer[], options?: LayerOptions): LayerGroup;
```

Comment:
Please add the `?` to `layers` here and remove the previous `layerGroup` declaration.

Reply:
Okay I feel like I'm going crazy I did think that this was previously giving a tslint error perhaps it was linting with an older version of typescript that didn't support it. Updated as suggested there is no need for overloads.

Reply:
@robert-prib-polestar The rule just means that you are not allow to do something like this:

```typescript
function firstIsOptional(options?: any, name: string): any {
  // ...
}
```
In other words, your non-optional parameters are not allowed after optional ones. This is just because of that:

```typescript
firstIsOptional(undefined, "Arne"); // Does not make sense...
secondIsOptional("Arne"); // Make sense...
```